pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -166,6 +165,8 @@ private MetadataCacheImpl(MetadataStore store, MetadataSerde<T> serde) {
                     T newValueObj;
                     byte[] newValue;
                     try {
+                        // Use clone and CAS zk to ensure thread safety
```

Comment:
We should do the same operation on the `readModifyUpdateOrCreate` method as well